services/project_service.rb

```diff
@@ -35,6 +37,8 @@ def create_project!(name: nil, repo_config: nil, enabled: nil, lane: nil, artifa
       project = self.project_data_source.create_project!(name: name, repo_config: repo_config, enabled: enabled, lane: lane, artifact_provider: artifact_provider)
       raise "Project couldn't be created" if project.nil?
       self.commit_repo_changes!(message: "Created project #{project.project_name}.")
+      # We shallow clone the repo to have the information needed for retrieving lanes.
+      _ = self.git_repo_for_project(project: project)
```

Comment:
let's put this into a method called `shallow_clone_project(project: nil)`
I want to make it even more obvious with code (not just comments) that we're relying on side effects here.